apps/users-service/src/address-book-service.service.ts

```diff
@@ -107,4 +113,33 @@ export class AddressBookService {
 
     return { status: 1 }
   }
+
+  async getAddressByPin (userId: string, pin: number): Promise<PinAddressI> {
+    try {
+      const getUser: UserI = await this.usersRepository.findOne({
+        _id: userId,
+        addressPin: pin
+      })
+
+      const addresses: AddressBookI[] = await this.repository.findAndPopulate(
```

Comment:
Add a filter to only query sharable address. That is why we added the shareable: boolean field in the address schema